### PR TITLE
Sneak peek: Recipe app

### PR DESCRIPTION
**Pull request by a-laarabi**
# What?
> I @a-laarabi and @tarek102, created the Recipe app. an app that keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and many more features. to check more information check [this](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md)

## data model
recipe_erd_2_members

# highlights:

- Created a [Kanban board](https://github.com/users/a-laarabi/projects/1/views/1) for task distribution
- Added Login page and registration page
- Added the Food list page to display a logged-in user's foods list
- Added the Recipe list page to display a logged-in user's recipes list
- Added a public recipes page to display a list of all public recipes ordered by the newest
- Added recipe details page to display details about a recipe
- Added General shopping list view
- Made sure there are no N+1 queries happening.
- Added a navigation menu that allows users to open all of the pages.
- Set up RSpec and Capybara testing

Capture d’écran 2022-11-17 à 14 30 08

### CODE REVIEW

File: app/views/shopping_lists/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <tbody>
+        <% @recipe.recipe_foods.each do |r| %>
+          <% if @food.include?(r.food) %>
+            <tr>
+              <th scope="row"><%= r.food.name %></th>
+              <td><%= r.quantity %> x <%= r.food.measurement_unit %></td>
+              <td>$<%= r.food.price * r.quantity %></td>
+            </tr>
+          <% end %>
+        <% end %>
```
Comment (codepantha) [inline app/views/shopping_lists/index.html.erb]: - [ ] The shopping list page should show the list of food that is missing for all recipes of the logged-in user. At the moment, it is simply showing all the food items in the recipe. Please fix this bug 🐛

File: spec/requests/foods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  describe 'GET foods#index' do
+    before(:each) do
+      user = User.create name: 'Tom', email: 'tom@example.com', password: '123456'
+      post user_session_path, params: { user: { email: user.email, password: user.password } }
+      get foods_path
+    end
+
+    it 'should return http success' do
+      expect(response).to have_http_status(:success)
+    end
```
Comment (codepantha) [inline spec/requests/foods_spec.rb:13]: - [ ] This test is currently failing as you can see in the attached image below 👇 Please try to fix it
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯
✔️ Descriptive PR
✔️ Used GitFlow
✔️ Beautiful UI
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/foods_spec.rb (hunk 2 above)
Comment (a-laarabi) [reply spec/requests/foods_spec.rb:13]: Heyy @codepantha thanks for your review we really appreciate it and we will do the necessary changes, however, I and my partner run rspec and we didn't have such an issue, you can see the result in the attached screenshot below 

Capture d’écran 2022-11-17 à 18 01 13

Comment (codepantha) [reply spec/requests/foods_spec.rb:13]: @a-laarabi, I apologize for the late reply. Please make the changes and request another review. If the tests pass, the next reviewer will be happy to approve this PR. Thanks 🙏

**Review by uwadonat — APPROVED** [review]
## Hi @a-laarabi,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Descriptive pull request message ✔️
- No linter errors ✔️

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**